pynfft/nfft.pxd

```diff
@@ -15,27 +15,39 @@
 # You should have received a copy of the GNU General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-from cnfft3 cimport nfft_plan
+from cnfft3 cimport nfft_plan, nfftf_plan, nfftl_plan
 
-cdef object nfft_flags_dict
-cdef object fftw_flags_dict
-cdef object nfft_supported_flags_tuple
 
-cdef class NFFT(object):
-    cdef nfft_plan _plan
-    cdef int _d
-    cdef int _M
-    cdef int _m
-    cdef object _N
-    cdef object _n
-    cdef object _dtype
-    cdef object _flags
-    cdef object _f   
-    cdef object _f_hat
-    cdef object _x
-    cdef void _precompute(self)
-    cdef void _trafo(self)
-    cdef void _trafo_direct(self)    
-    cdef void _adjoint(self)
-    cdef void _adjoint_direct(self)
+cdef:
+    object nfft_flags_dict
+    object fftw_flags_dict
+    object nfft_supported_flags_tuple
+
+
+ctypedef union _nfft_plan:
```

Comment:
nice. good way to avoid too much duplication

Reply:
If you know a good way to avoid some of the other duplication as well, please let me know.

Reply:
> If I understand correctly any given NFFT plan will only have one of these actually initialized via a call to `nfft_init_guru`, etc.

Yes (or rather, the single memory location will be initialized, and only one interpretation will be correct at a given time).